src/mlpack/core/optimizers/cmaes/cmaes_impl.hpp

```diff
@@ -0,0 +1,735 @@
+      :
+        function(function),
+        N(-1),
+        typicalXcase(false),
+        stopMaxFunEvals(-1),
+        facmaxeval(1.0),
+        stopTolFunHist(1e-14),
+        stopTolX(0),
+        stopTolUpXFactor(1e3),
+        lambda(-1),
+        mu(-1),
+        mucov(-1),
+        mueff(-1),
+        damps(-1),
+        cs(-1),
+        ccumcov(-1),
+        ccov(-1),
+        facupdateCmode(1),
+        weightMode(UNINITIALIZED_WEIGHTS)
+  {
+    stStopFitness.flg = false;
+    stStopFitness.val = -std::numeric_limits<double>::max();
+    updateCmode.modulo = -1;
+    updateCmode.maxtime = -1;
+
+     N = function.NumFunctions();
+    if ( N <= 0)
+      throw std::runtime_error("Problem dimension N undefined.");
+    bool startP  = true;
+    bool initDev = true;
+
+    for (int i=0; i<N; i++)
+    {
+     if (start[i]   < 1.0e-200) startP  = false;
+     if (stdDivs[i] < 1.0e-200) initDev = false;
+    }
+
+     if (evalDiff = 1e-14) stopTolFun = 1e-14;
+     else
+        stopTolFun = evalDiff;
+
+if (!startP)
+Log::Warn << " WARNING: initial start point undefined." <<
+"Please specify if incorrect results detected."
+<< "DEFAULT = 0.5...0.5." << std::endl;
+
+if (!initDev)
+Log::Warn << "WARNING: initialStandardDeviations undefined."
+<< " Please specify if incorrect results detected. DEFAULT = 0.3...0.3."
+<< std::endl;
+
+    if (weightMode == UNINITIALIZED_WEIGHTS)
+      weightMode = LOG_WEIGHTS;
+
+    diagonalCov = 0; // default is 0, but this might change in future
+
+      xstart.set_size(N);
+      if (startP)
+      {
+        for (int i = 0; i < N; ++i) xstart[i] = start[i];
+      }
+     else
+      {
+        typicalXcase = true;
+        for (int i = 0; i < N; i++) xstart[i] = 0.5;
+      }
+
+    rgInitialStds.set_size(N);
+    if (initDev)
+      {
+        for (int i = 0; i < N; ++i) rgInitialStds[i] = stdDivs[i];
+      }
+      else
+      {
+        for (int i = 0; i < N; ++i) rgInitialStds[i] = double(0.3);
+      }
+
+    if (lambda < 2)
+      lambda = 4 + (int) (3.0*log((double) N));
+    if (mu <= 0)
+      mu = lambda / 2;
+
+      weights.set_size(mu);
+      switch (weightMode)
+      {
+      case LINEAR_WEIGHTS:
+        for (int i = 0; i < mu; ++i) weights[i] = mu - i;
+        break;
+      case EQUAL_WEIGHTS:
+        for (int i = 0; i < mu; ++i) weights[i] = 1;
+        break;
+      case LOG_WEIGHTS:
+      default:
+        for (int i = 0; i < mu; ++i) weights[i] = log(mu + 1.) - log(i + 1.);
+        break;
+      }
+
+      // normalize weights vector and set mueff
+      double s1 = 0, s2 = 0;
+      for (int i = 0; i < mu; ++i)
+      {
+        s1 += weights[i];
+        s2 += weights[i]*weights[i];
+      }
+      mueff = s1*s1/s2;
+      for (int i = 0; i < mu; ++i)
+        weights[i] /= s1;
+
+      if (mu < 1 || mu > lambda || (mu == lambda
+        && weights[0] == weights[mu - 1]))
+      throw std::runtime_error("setWeights(): invalid setting of mu or lambda");
+
+    if (cs > 0)
```

Comment:
as now mu and lambda cannot be set by user. Ive removed this comment.
it was based on the previous version.
thanks for pointing this out 

Done